src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -24,19 +24,21 @@
  */
 package org.spongepowered.api.plugin;
 
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.util.regex.Pattern;
 
 /**
  * An annotation used to describe and mark a Sponge plugin.
  */
-@Target(TYPE)
-@Retention(RUNTIME)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
 public @interface Plugin {
 
+    Pattern ID_PATTERN = Pattern.compile("[a-z0-9-_]+");
```

Comment:
I'm not sure if it's a good idea to allow to use parentheses for the plugin ID. I could see this easily cause trouble if they're used as identification and would be used for example in the URL of the project on Ore, this would result in having to encode them. In my opinion it should be only the characters I mentioned above.

Requiring the first character to be a letter kind of makes sense though, I'll add that. And not sure about requiring lower case letters only but IMO it would kind of make sense so they're consistent and not every second plugin uses capitalized names and the others lower case names...